tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,150 @@
+import copy
+import pytest
+
+from configuration import available_ports, ALL_TEST_CIPHERS, ALL_TEST_CURVES, MINIMAL_TEST_CERTS
+from common import Ciphers, ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process  # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version, to_bytes
+
+PADDING_SIZE_MIN = 0
+PADDING_SIZE_SMALL = 250
+PADDING_SIZE_MEDIUM = 1000
+PADDING_SIZE_MAX = 1 << 14
+
+PADDING_SIZES = [
+    PADDING_SIZE_MIN,
+    PADDING_SIZE_SMALL,
+    PADDING_SIZE_MEDIUM,
+    PADDING_SIZE_MAX
+]
+
+PAYLOAD_SIZE_SMALL = 10
+PAYLOAD_SIZE_MEDIUM = 100
+# openssl has a max fragment length of 4096 bytes
+PAYLOAD_SIZE_LARGE = 4096
+
+OPENSSL_PAYLOAD_SIZES = [PAYLOAD_SIZE_SMALL,
+                         PAYLOAD_SIZE_MEDIUM, PAYLOAD_SIZE_LARGE]
+
+# https://www.rfc-editor.org/rfc/rfc5116
+TLS13_CIPHERS_WITH_16_BYTE_TAGS = [
+    Ciphers.AES128_GCM_SHA256,
+    Ciphers.AES256_GCM_SHA384,
+    Ciphers.CHACHA20_POLY1305_SHA256
+]
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS_WITH_16_BYTE_TAGS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+# only tls 1.3 supports record padding
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("padded_block_size", PADDING_SIZES, ids=get_parameter_name)
+@pytest.mark.parametrize("payload_size", OPENSSL_PAYLOAD_SIZES, ids=get_parameter_name)
+def test_s2n_13_server_handles_padded_records(managed_process, cipher, provider, curve, protocol, certificate,
+                                              padded_block_size, payload_size):
```

Comment:
```suggestion
                                              padding_size, payload_size):
```
The name 'padded_block_size' is kind of misleading, a padded block is something we usually associate with ciphers doing encryption/decryption. Record padding is something different and we shouldn't mix up the two concepts.